### DISCUSSION TITLE
Updating multiple slices

### DISCUSSION THREAD

**Discussion by ithilbor** [pmndrs/zustand#1602] [Q&A] [answered]
There is a version of this function but using Typescript?

```node
import { createBearSlice } from './bearSlice'
import { createFishSlice } from './fishSlice'

export const createBearFishSlice = (set) => ({
  addBearAndFish: () => {
    createBearSlice(set).addBear()
    createFishSlice(set).addFish()
  },
})
```

Because in my code I'm doing:

```node
export const createBearFishSlice: StateCreator<BearFishSliceT & BearSliceT & FishSliceT, [], [], BearFishSliceT> = (set, get) => ({
    addFishAndBear: (fishes, bears) => {
      createBearSlice(set, get, createBearSlice).addBear(bears);
      createFishSlice(set, get, createFishSlice).addFish(fishes);
    },
    anotherFun: () => {},
  });
```
  
And I get this error for both stores declared inside the function "addFishAndBear":
  
"The argument of type 'StateCreator<BearSliceT, [], [], BearSliceT>' is not assignable to the type StoreApi BearSliceT'.ts(2345)"
 
Also if I don't specify the get and store in the parameters and I write: "createBearSlice(set).addBear(bears);" i get this error:
"You need to specify 3 arguments but you specified only one.ts(2554)
vanilla.d.ts(17, 208): No argument specified for 'getState'."
  
  Do you have any idea on how to solve it? Or if I'm doing something wrong? Thanks in advance!

**Comment by dai-shi** [comment]
It's pretty hard to type multiple slices in TypeScript, but maybe possible.
Can you create a reproduction with https://www.typescriptlang.org/play/ ? Put all code (`import` works) in it and create a shortcut link (see plugins).

**Reply by ithilbor** [reply]
Of curse I can, I'will do it right now thank you @dai-shi !

**Reply by ithilbor** [reply]
Here it is the reproduction link @dai-shi : https://tsplay.dev/WK94oN

**Comment by dai-shi** [comment, marked as answer]
Here you go: https://tsplay.dev/WJ7vZw

**Reply by ithilbor** [reply]
Noo I missed to specify the api, my bad, thanks a lot @dai-shi!

**Reply by dai-shi** [reply]
You are welcome!
